solidity/random-beacon/contracts/libraries/Relay.sol

```diff
@@ -133,23 +127,36 @@ library Relay {
             "Invalid entry"
         );
 
-        // If the soft timeout has been exceeded apply stake slashing for
-        // all group members. Note that `getSlashingFactor` returns the
-        // factor multiplied by 1e18 to avoid precision loss. In that case
-        // the final result needs to be divided by 1e18.
-        slashingAmount =
-            (getSlashingFactor(self, dkgGroupSize) *
-                self.relayEntrySubmissionFailureSlashingAmount) /
-            1e18;
-
         emit RelayEntrySubmitted(self.currentRequestID, msg.sender, entry);
 
         self.previousEntry = AltBn128.g1Unmarshal(entry);
         self.currentRequestID = 0;
         self.currentRequestGroupID = 0;
         self.currentRequestStartBlock = 0;
+    }
+
+    /// @notice Calculates the slashing amount for all group members.
+    /// @dev Must be used when a soft timeout was hit.
+    /// @return Amount by which group members should be slashed
+    ///         in case the relay entry was submitted after the soft timeout.
+    function calculateSlashingAmount(Data storage self)
+        internal
+        returns (uint256)
+    {
+        uint256 softTimeoutBlock = self.currentRequestStartBlock +
+            (dkgGroupSize * self.relayEntrySubmissionEligibilityDelay);
+
+        uint256 submissionDelay = block.number - softTimeoutBlock;
```

Comment:
Previously we had `if (block.number > softTimeoutBlock) {` check. Don't we need it anymore?

Reply:
Since we started paying closer attention to gas usage I think it would be a duplicate check. This is an internal function that is called only from `submitRelayEntry()` which already checks if the soft timeout was hit. There's also a comment that warns this function must be called only after a soft timeout.

Reply:
Sure, sounds good 0d54f43